src/features/claims/claim-details/components/ClaimRestrictionInformation.tsx

```diff
@@ -0,0 +1,116 @@
+import styled from '@emotion/styled'
+import { Button, Flex, Shadowed } from '@hedvig-ui'
+import { useConfirmDialog } from '@hedvig-ui/Modal/use-confirm-dialog'
+import chroma from 'chroma-js'
+import { ResourceAccessOverview } from 'features/resource-access/overview/ResourceAccessOverview'
+import React, { useState } from 'react'
+import { ShieldLockFill } from 'react-bootstrap-icons'
+import { toast } from 'react-hot-toast'
+import {
+  ClaimPageDocument,
+  ClaimPageQuery,
+  ResourceAccessInformation,
+  useReleaseResourceAccessMutation,
+} from 'types/generated/graphql'
+
+const Subtext = styled.span`
+  font-size: 0.85rem;
+  margin-top: 0.2rem;
+  color: ${({ theme }) =>
+    chroma(theme.foreground)
+      .alpha(0.7)
+      .hex()};
+`
+
+export const ClaimRestrictionInformation: React.FC<{
+  restriction: ResourceAccessInformation
+  claimId: string
+}> = ({ restriction, claimId }) => {
+  const [showModal, setShowModal] = useState(false)
+  const { confirm } = useConfirmDialog()
+  const [releaseResourceAccess] = useReleaseResourceAccessMutation()
+
+  const handleReleaseAccess = () => {
+    toast.promise(
+      releaseResourceAccess({
+        variables: { resourceId: claimId },
+        update: (cache) => {
+          const cachedData = cache.readQuery({
+            query: ClaimPageDocument,
+            variables: { claimId },
+          }) as ClaimPageQuery
+
+          cache.writeQuery({
+            query: ClaimPageDocument,
+            data: {
+              claim: {
+                __typename: 'Claim',
+                ...cachedData.claim,
+                restriction: null,
+              },
+            },
+          })
+        },
+      }),
+      {
+        loading: 'Removing restriction',
+        success: 'Restriction removed',
+        error: 'Could not remove restriction',
+      },
+    )
+  }
+
+  return (
+    <>
+      {showModal && (
+        <ResourceAccessOverview
+          onClose={() => setShowModal(false)}
+          resourceId={claimId}
+        />
+      )}
+      <Flex direction="row" align="center">
```

Comment:
Why not inside `Flex` component? Like in bottom maybe?
```suggestion
      <Flex direction="row" align="center">
         // ...
         {showModal && <Modal />}
      </Flex>
```

Reply:
I thought Modals were `absolute` positioned? 🤔 I just thought it made sense to put it as high up as possible, since it will be rendered like that anyway 😅